packages/cli/src/commands/transfer/erc20.ts

```diff
@@ -34,28 +31,53 @@ export default class TransferErc20 extends BaseCommand {
   ]
 
   async run() {
-    const kit = await this.getKit()
     const res = await this.parse(TransferErc20)
+    const client = await this.getPublicClient()
+    const wallet = await this.getWalletClient()
+
+    if (!wallet) {
+      throw new Error('TODO: only AKV doesnt return a wallet')
+    }
 
     const from = res.flags.from
     const to = res.flags.to
-    const value = new BigNumber(res.flags.value)
+    const value = res.flags.value
+    const feeCurrency = res.flags.gasCurrency
+
+    const erc20Contract = {
+      abi: erc20Abi,
+      address: res.flags.erc20Address,
+      ...(feeCurrency ? { feeCurrency } : {}),
+    } as const
 
-    kit.defaultAccount = from
-    let celoToken: Erc20Wrapper<IERC20>
+    let decimals: number
     try {
-      celoToken = await kit.contracts.getErc20(res.flags.erc20Address)
-      // this call allow us to check if it is a valid erc20
-      await celoToken.balanceOf(res.flags.from)
+      decimals = await client.readContract({
+        ...erc20Contract,
+        functionName: 'decimals',
+        args: [],
+      })
     } catch {
       failWith('Invalid erc20 address')
     }
     await newCheckBuilder(this)
       .isNotSanctioned(from)
       .isNotSanctioned(to)
-      .hasEnoughErc20(from, value, res.flags.erc20Address)
+      .isValidWalletSigner(from)
+      .hasEnoughErc20(from, value, res.flags.erc20Address, decimals)
       .runChecks()
 
-    await displaySendTx('transfer', celoToken.transfer(to, value.toFixed()))
+    await displaySendViemContractCall(
+      'ERC20',
```

Comment:
you know erc20 contracts have a `symbol` and a `name` methods. 

can we use one of them to display what token is being transfered?

Reply:
good idea